lib/repo/labels.js

```diff
@@ -0,0 +1,301 @@
+    slug: toLabelSlug(d.name)
+  }))
+
+  const unknown = []
+  const remove = []
+  const actual = []
+  const toggle = []
+  const change = []
+
+  prev
+    .filter(d => !expected.find(e => d.slug === e.slug))
+    .forEach(d => {
+      if (own.call(replaceMap, d.slug)) {
+        const name = replaceMap[d.slug]
+        const replace = expected.find(e => e.slug === name)
+        const renameable =
+          !prev.find(e => e.slug === name) &&
+          !change.find(e => e.to.slug === name)
+        const list = renameable ? change : toggle
+
+        list.push({from: d, to: replace})
+      } else if (ghLabels.remove.includes(d.slug)) {
+        remove.push(d)
+      } else {
+        unknown.push(d)
+      }
+    })
+
+  // To do: add a scheme, e.g., `component/`, that’s free to use.
+  unknown.forEach(d => {
+    console.log(
+      '  ' + chalk.blue('ℹ') + ' unknown label ' + chalk.blue('%s'),
+      d.name
+    )
+  })
+
+  const add = expected
+    // Not already existing.
+    .filter(d => !prev.find(e => d.slug === e.slug))
+    // Not marked for a change.
+    .filter(d => !change.find(e => d.slug === e.to.slug))
+
+  if (add.length !== 0) {
+    const res = await ghQuery(
+      []
+        .concat(
+          'mutation {',
+          add.map(
+            (d, i) =>
+              '  create' +
+              i +
+              ': createLabel(input: { repositoryId: "' +
+              repositoryId +
+              '", name: "' +
+              d.name +
+              '", description: "' +
+              d.description +
+              '", color: "' +
+              d.color +
+              '" }) {\n    label { id }\n  }'
+          ),
+          '}'
+        )
+        .join('\n'),
+      {headers: {accept: labelsPreviewAccept}}
+    )
+
+    add.forEach((d, i) => {
+      actual.push({...d, id: res['create' + i].label.id})
+
+      console.log(
+        '  ' + chalk.green('✓') + ' created label %s',
+        chalk.blue(d.name)
+      )
+    })
+  }
+
+  prev
+    // Not added above.
+    .filter(d => !add.find(e => d.slug === e.slug))
+    .forEach(ac => {
+      const ex = expected.find(e => ac.slug === e.slug)
+
+      // Ignore unexpected values.
+      if (!ex) {
+        return
+      }
+
+      if (
+        ac.name === ex.name &&
+        ac.description === ex.description &&
+        ac.color === ex.color
+      ) {
+        actual.push(ac)
+      } else {
+        change.push({from: ac, to: ex})
+      }
+    })
+
+  if (change.length !== 0) {
+    await ghQuery(
+      []
+        .concat(
+          'mutation {',
+          change.map(
+            (d, i) =>
+              '  change' +
+              i +
+              ': updateLabel(input: { id: "' +
+              d.from.id +
+              '", name: "' +
+              d.to.name +
+              '", description: "' +
+              d.to.description +
+              '", color: "' +
+              d.to.color +
+              '" }) {\n    clientMutationId\n  }'
```

Comment:
could this be a good place to use a [template string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)?